[2025-07-25T11:10:26.217+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-07-25T11:10:26.229+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-10T00:00:00+00:00 [queued]>
[2025-07-25T11:10:26.238+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-10T00:00:00+00:00 [queued]>
[2025-07-25T11:10:26.238+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-07-25T11:10:26.251+0000] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): extract_task.crawl_news> on 2025-07-10 00:00:00+00:00
[2025-07-25T11:10:26.259+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'elt_to_dwh', 'extract_task.crawl_news', 'scheduled__2025-07-10T00:00:00+00:00', '--job-id', '71', '--raw', '--subdir', 'DAGS_FOLDER/elt_to_dwh.py', '--cfg-path', '/tmp/tmpy487d2yz']
[2025-07-25T11:10:26.261+0000] {standard_task_runner.py:105} INFO - Job 71: Subtask extract_task.crawl_news
[2025-07-25T11:10:26.263+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=2709) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-25T11:10:26.264+0000] {standard_task_runner.py:72} INFO - Started process 2719 to run task
[2025-07-25T11:10:26.296+0000] {task_command.py:467} INFO - Running <TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-10T00:00:00+00:00 [running]> on host 74cfddf97016
[2025-07-25T11:10:26.360+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='elt_to_dwh' AIRFLOW_CTX_TASK_ID='extract_task.crawl_news' AIRFLOW_CTX_EXECUTION_DATE='2025-07-10T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-07-10T00:00:00+00:00'
[2025-07-25T11:10:26.361+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-07-25T11:10:26.375+0000] {logging_mixin.py:190} INFO - Get news from 20250710T0000 to 20250710T2359
[2025-07-25T11:10:29.616+0000] {logging_mixin.py:190} INFO - Get news from 20250710T0000 to 20250710T1200
[2025-07-25T11:10:31.894+0000] {logging_mixin.py:190} INFO - Get news from 20250710T1201 to 20250710T2359
[2025-07-25T11:10:34.600+0000] {logging_mixin.py:190} INFO - The process of crawling 2344 news was successful
[2025-07-25T11:10:34.601+0000] {logging_mixin.py:190} INFO - Saving at /opt/airflow/data/raw/news/crawl_news-20250710.json
[2025-07-25T11:10:34.605+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-07-25T11:10:34.616+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-07-25T11:10:34.617+0000] {taskinstance.py:353} INFO - Marking task as SUCCESS. dag_id=elt_to_dwh, task_id=extract_task.crawl_news, run_id=scheduled__2025-07-10T00:00:00+00:00, execution_date=20250710T000000, start_date=20250725T111026, end_date=20250725T111034
[2025-07-25T11:10:34.662+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-07-25T11:10:34.691+0000] {taskinstance.py:3901} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-07-25T11:10:34.699+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-07-25T11:50:06.218+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-07-25T11:50:06.231+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-10T00:00:00+00:00 [queued]>
[2025-07-25T11:50:06.239+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-10T00:00:00+00:00 [queued]>
[2025-07-25T11:50:06.239+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-07-25T11:50:06.253+0000] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): extract_task.crawl_news> on 2025-07-10 00:00:00+00:00
[2025-07-25T11:50:06.264+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'elt_to_dwh', 'extract_task.crawl_news', 'scheduled__2025-07-10T00:00:00+00:00', '--job-id', '142', '--raw', '--subdir', 'DAGS_FOLDER/elt_to_dwh.py', '--cfg-path', '/tmp/tmpkju2rmb1']
[2025-07-25T11:50:06.278+0000] {standard_task_runner.py:105} INFO - Job 142: Subtask extract_task.crawl_news
[2025-07-25T11:50:06.287+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=4460) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-25T11:50:06.288+0000] {standard_task_runner.py:72} INFO - Started process 4470 to run task
[2025-07-25T11:50:06.330+0000] {task_command.py:467} INFO - Running <TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-10T00:00:00+00:00 [running]> on host 74cfddf97016
[2025-07-25T11:50:06.408+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='elt_to_dwh' AIRFLOW_CTX_TASK_ID='extract_task.crawl_news' AIRFLOW_CTX_EXECUTION_DATE='2025-07-10T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-07-10T00:00:00+00:00'
[2025-07-25T11:50:06.410+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-07-25T11:50:06.436+0000] {logging_mixin.py:190} INFO - Get news from 20250710T0000 to 20250710T2359
[2025-07-25T11:50:07.286+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/scripts/elt_to_dwh/extract/crawl_news.py", line 40, in crawl_news
    raise KeyError(f"'feed' key not found in response. Full response:\n{json.dumps(response, indent=4)}")
KeyError: '\'feed\' key not found in response. Full response:\n{\n    "Information": "We have detected your API key as RSKHO6B8SELFWUYG and our standard API rate limit is 25 requests per day. Please subscribe to any of the premium plans at https://www.alphavantage.co/premium/ to instantly remove all daily rate limits."\n}'
[2025-07-25T11:50:07.297+0000] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=elt_to_dwh, task_id=extract_task.crawl_news, run_id=scheduled__2025-07-10T00:00:00+00:00, execution_date=20250710T000000, start_date=20250725T115006, end_date=20250725T115007
[2025-07-25T11:50:07.325+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-07-25T11:50:07.326+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 142 for task extract_task.crawl_news ('\'feed\' key not found in response. Full response:\n{\n    "Information": "We have detected your API key as RSKHO6B8SELFWUYG and our standard API rate limit is 25 requests per day. Please subscribe to any of the premium plans at https://www.alphavantage.co/premium/ to instantly remove all daily rate limits."\n}'; 4470)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/scripts/elt_to_dwh/extract/crawl_news.py", line 40, in crawl_news
    raise KeyError(f"'feed' key not found in response. Full response:\n{json.dumps(response, indent=4)}")
KeyError: '\'feed\' key not found in response. Full response:\n{\n    "Information": "We have detected your API key as RSKHO6B8SELFWUYG and our standard API rate limit is 25 requests per day. Please subscribe to any of the premium plans at https://www.alphavantage.co/premium/ to instantly remove all daily rate limits."\n}'
[2025-07-25T11:50:07.349+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-07-25T11:50:07.370+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-25T11:50:07.379+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-07-26T04:56:14.713+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-07-26T04:56:14.729+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-10T00:00:00+00:00 [queued]>
[2025-07-26T04:56:14.738+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-10T00:00:00+00:00 [queued]>
[2025-07-26T04:56:14.739+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-07-26T04:56:14.755+0000] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): extract_task.crawl_news> on 2025-07-10 00:00:00+00:00
[2025-07-26T04:56:14.761+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'elt_to_dwh', 'extract_task.crawl_news', 'scheduled__2025-07-10T00:00:00+00:00', '--job-id', '18', '--raw', '--subdir', 'DAGS_FOLDER/elt_to_dwh.py', '--cfg-path', '/tmp/tmp18cj0n90']
[2025-07-26T04:56:14.764+0000] {standard_task_runner.py:105} INFO - Job 18: Subtask extract_task.crawl_news
[2025-07-26T04:56:14.765+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=545) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-26T04:56:14.765+0000] {standard_task_runner.py:72} INFO - Started process 555 to run task
[2025-07-26T04:56:14.802+0000] {task_command.py:467} INFO - Running <TaskInstance: elt_to_dwh.extract_task.crawl_news scheduled__2025-07-10T00:00:00+00:00 [running]> on host 3bc71adbf7e0
[2025-07-26T04:56:14.866+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='elt_to_dwh' AIRFLOW_CTX_TASK_ID='extract_task.crawl_news' AIRFLOW_CTX_EXECUTION_DATE='2025-07-10T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-07-10T00:00:00+00:00'
[2025-07-26T04:56:14.866+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-07-26T04:56:14.880+0000] {logging_mixin.py:190} INFO - Get news from 20250710T0000 to 20250710T2359
[2025-07-26T04:56:17.696+0000] {logging_mixin.py:190} INFO - Get news from 20250710T0000 to 20250710T1200
[2025-07-26T04:56:19.865+0000] {logging_mixin.py:190} INFO - Get news from 20250710T1201 to 20250710T2359
[2025-07-26T04:56:23.105+0000] {logging_mixin.py:190} INFO - The process of crawling 2344 news was successful
[2025-07-26T04:56:23.105+0000] {logging_mixin.py:190} INFO - Saving at /opt/airflow/data/raw/news/crawl_news-20250710.json
[2025-07-26T04:56:23.108+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-07-26T04:56:23.112+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-07-26T04:56:23.112+0000] {taskinstance.py:353} INFO - Marking task as SUCCESS. dag_id=elt_to_dwh, task_id=extract_task.crawl_news, run_id=scheduled__2025-07-10T00:00:00+00:00, execution_date=20250710T000000, start_date=20250726T045614, end_date=20250726T045623
[2025-07-26T04:56:23.169+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-07-26T04:56:23.182+0000] {taskinstance.py:3901} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-07-26T04:56:23.187+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
